trunk/source/IcfgBuilder/src/de/uni_freiburg/informatik/ultimate/plugins/generator/icfgbuilder/cfg/CfgBuilder.java

```diff
@@ -1073,29 +1074,42 @@ private BoogieIcfgLocation buildBreak(final BoogieIcfgLocation currentLocation,
 		}
 
 		private BoogieIcfgLocation buildLabel(final IIcfgElement currentElement, final Label st) {
-			final BoogieIcfgLocation newLocation = getLocNodeForLabel(new StringDebugIdentifier(st.getName()), st);
-			final BoogieIcfgLocation resultLocation;
+			final BoogieIcfgLocation newLoc = getLocNodeForLabel(new StringDebugIdentifier(st.getName()), st);
+			final BoogieIcfgLocation resultLoc;
 			if (currentElement instanceof BoogieIcfgLocation) {
 				// We do not want to introduce a new program point for this label but
 				// merge the just constructed BoogieIcfgLocation with currentElement.
 				// A merge in the other direction is not possible, because currentElement
 				// might be a the successor of an if-then-else. If we replace currentElement
 				// here, we also would have to replace it for the other branches.
-				mergeLocNodes(newLocation, (BoogieIcfgLocation) currentElement, true);
-				resultLocation = (BoogieIcfgLocation) currentElement;
+				mergeLocNodes(newLoc, (BoogieIcfgLocation) currentElement, true);
+				resultLoc = (BoogieIcfgLocation) currentElement;
 			} else {
-				endStatementSequence((StatementSequence) currentElement, newLocation);
-				if (!isAuxiliaryLabel(st)) {
-					// TODO: add to labels of CFG after we have labels in the CFG
-				}
-				resultLocation = newLocation;
+				endStatementSequence((StatementSequence) currentElement, newLoc);
+				resultLoc = newLoc;
+			}
+			if (!isAuxiliaryLabel(st)) {
+				mIcfg.getLocationsOfInterest().add(resultLoc);
 			}
-			return resultLocation;
+			return resultLoc;
 		}
 
-		// TODO Implement support for the attribute
+		/**
+		 * Auxiliary labels identified by an {@link NamedAttribute}.
+		 */
 		private boolean isAuxiliaryLabel(final Label st) {
-			return true;
+			if (st.getAttributes() == null) {
+				return false;
+			}
+			for (final NamedAttribute attr : st.getAttributes()) {
+				if (attr.getName().equals(BoogieUtils.AUXILIARY_LABEL)) {
+					if (attr.getValues().length != 0) {
+						throw new AssertionError("Attribut must not have values");
```

Comment:
nitpick: `Attibute`. Also, it seems slightly strange to check this here, shouldn't such cases be caught earlier? (not a blocker though)

Reply:
I am not sure if I understand.
It would definitely be more professional, 
* if we would have some code that checks at the beginning of the IcfgBuilder if the Boogie Code meets our expections
* if these expections would be documented somewhere.

However, we would need someone who can write and maintain this documentation and code.

Reply:
I was thinking of something like the `VALIDATORS` mechanism in `BoogieASTNode`, which would throw an exception already when the nodes are created (independently of the `IcfgBuilder` plugin). But upon reflection, this may not be the right choice for attributes.